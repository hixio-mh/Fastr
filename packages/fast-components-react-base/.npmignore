# Tests
__test__/
*.spec.*
*.test.*

# Source files
app/
coverage/
src/
www/

# babel config
babel.config.js
tsconfig.json
tslint.json
webpack.config.js