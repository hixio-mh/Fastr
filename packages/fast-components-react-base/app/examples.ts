import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CardExamples from "../src/card/examples.data";
export { CardExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import DialogExamples from "../src/dialog/examples.data";
export { DialogExamples };

import DividerExamples from "../src/divider/examples.data";
export { DividerExamples };

import HorizontalOverflowExamples from "../src/horizontal-overflow/examples.data";
export { HorizontalOverflowExamples };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import RadioExamples from "../src/radio/examples.data";
export { RadioExamples };

import TabsExamples from "../src/tabs/examples.data";
export { TabsExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
