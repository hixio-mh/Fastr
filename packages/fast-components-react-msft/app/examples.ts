import ButtonExamples from "../src/button/examples.data";
export { ButtonExamples };

import CallToActionExamples from "../src/call-to-action/examples.data";
export { CallToActionExamples };

import CaptionExamples from "../src/caption/examples.data";
export { CaptionExamples };

import CheckboxExamples from "../src/checkbox/examples.data";
export { CheckboxExamples };

import Dialog from "../src/dialog/examples.data";
export { Dialog };

import DividerExample from "../src/divider/examples.data";
export { DividerExample };

import FlipperExample from "../src/flipper/examples.data";
export { FlipperExample };

import HeadingExample from "../src/heading/examples.data";
export { HeadingExample };

import HypertextExamples from "../src/hypertext/examples.data";
export { HypertextExamples };

import ImageExamples from "../src/image/examples.data";
export { ImageExamples };

import LabelExamples from "../src/label/examples.data";
export { LabelExamples };

import MetatextExamples from "../src/metatext/examples.data";
export { MetatextExamples };

import ParagraphExamples from "../src/paragraph/examples.data";
export { ParagraphExamples };

import ProgressExamples from "../src/progress/examples.data";
export { ProgressExamples };

import SubheadingExamples from "../src/subheading/examples.data";
export { SubheadingExamples };

import ToggleExamples from "../src/toggle/examples.data";
export { ToggleExamples };

import TextFieldExamples from "../src/text-field/examples.data";
export { TextFieldExamples };

import TypographyExamples from "../src/typography/examples.data";
export { TypographyExamples };
