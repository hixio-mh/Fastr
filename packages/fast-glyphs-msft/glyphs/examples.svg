<svg width="16" height="16" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="Canvas">
<g id="Group">
<g id="Vector">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.3 5.8L 2.3 3.9L 0.6 4.8L -3.8147e-07 3.8L 1.7 2.9L 0.1 2L 0.7 1L 2.2 1.9L 2.2 2.38419e-08L 3.3 2.38419e-08L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(10.3999 1.09998)"/>
</g>
<g id="Vector_2">
<path d="M 4 2.9L 5.6 3.8L 5 4.8L 3.4 3.9L 3.4 5.8L 2.2 5.8L 2.2 3.9L 0.6 4.8L 4.76837e-08 3.8L 1.6 2.9L 4.76837e-08 2L 0.6 1L 2.2 1.9L 2.2 -1.90735e-07L 3.3 -1.90735e-07L 3.3 1.9L 4.9 1L 5.5 2L 4 2.9Z" transform="translate(1.19995 10.2)"/>
</g>
<g id="Vector_3">
<path d="M 0 8L 0 0L 8 0L 8 8L 0 8ZM 1 1L 1 7L 7 7L 7 1L 1 1ZM 14.4 13.1L 16 14L 15.4 15L 13.8 14.1L 13.8 16L 12.7 16L 12.7 14.1L 11 15.1L 10.4 14.1L 12 13.2L 10.4 12.3L 11 11.3L 12.6 12.2L 12.6 10.3L 13.7 10.3L 13.7 12.2L 15.3 11.3L 15.9 12.3L 14.4 13.1Z"/>
</g>
</g>
</g>
</svg>