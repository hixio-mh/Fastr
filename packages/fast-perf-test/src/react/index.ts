import React from "react";
import ReactDOM from "react-dom";
import { Checkbox, TextField } from "@microsoft/fast-components-react-msft";

const root = document.createElement("div");
document.body.appendChild(root);

function getQueryVariable(variable) {
    var query = window.location.search.substring(1);
    var vars = query.split("&");
    for (var i = 0; i < vars.length; i++) {
        var pair = vars[i].split("=");
        if (decodeURIComponent(pair[0]) == variable) {
            return decodeURIComponent(pair[1]);
        }
    }
}

const count = getQueryVariable("count");

function renderItems() {
    return React.createElement(
        React.Fragment,
        {},
        new Array(typeof count === "string" ? (count as any) * 1 : 2000)
            .fill(0)
            .map((item, index) => React.createElement(TextField))
    );
}

function render(): void {
    ReactDOM.render(React.createElement(renderItems), root);
}

render();
