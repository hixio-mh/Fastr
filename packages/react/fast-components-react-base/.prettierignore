coverage/*
dist/*
www/*
src/**/.tmp/