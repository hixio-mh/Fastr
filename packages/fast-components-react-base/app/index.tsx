import * as React from "react";
import * as ReactDOM from "react-dom";
import Site, {
    componentFactory,
    formChildFromExamplesFactory,
    IFormChildOption,
    ISiteProps,
    SiteCategory,
    SiteCategoryIcon,
    SiteCategoryItem,
    SiteTitle,
    SiteTitleBrand
} from "@microsoft/fast-development-site-react";
import * as examples from "./examples";

import {
    Tab,
    TabItem,
    TabPanel
} from "../src";
import TabSchema from "../src/tabs/tab.schema.json";
import TabItemSchema from "../src/tabs/tab-item.schema.json";
import TabPanelSchema from "../src/tabs/tab-panel.schema.json";

/**
 * Create the root node
 */
const root: HTMLElement = document.createElement("div");
root.setAttribute("id", "root");
document.body.appendChild(root);

let formChildOptions: IFormChildOption[] = formChildFromExamplesFactory(examples);

formChildOptions = formChildOptions.concat([
    {
        name: "Tab",
        component: Tab,
        schema: TabSchema
    },
    {
        name: "TabItem",
        component: TabItem,
        schema: TabItemSchema
    },
    {
        name: "TabPanel",
        component: TabPanel,
        schema: TabPanelSchema
    }
]);

/* tslint:disable */
function render(): void {
    ReactDOM.render(
        <Site formChildOptions={formChildOptions} showTransparencyToggle={true}>
            <SiteTitle slot={"title"}>
                <SiteTitleBrand>FAST</SiteTitleBrand> base component documentation
            </SiteTitle>
            <SiteCategory slot={"category"} name={"Components"}>
                {componentFactory(examples)}
            </SiteCategory>
        </Site>,
        root
    );
}

render();
