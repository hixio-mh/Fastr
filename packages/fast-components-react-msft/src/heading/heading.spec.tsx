import * as React from "react";
import * as Adapter from "enzyme-adapter-react-16";
import { configure, mount, shallow } from "enzyme";
import examples from "./examples.data";
import {
    generateSnapshots,
    SnapshotTestSuite,
} from "@microsoft/fast-jest-snapshots-react";
import MSFTHeading, {
    HeadingHandledProps,
    HeadingSize,
    HeadingTag,
    HeadingUnhandledProps,
} from "./heading";
import { Typography } from "@microsoft/fast-components-react-base";
import { Heading, HeadingProps } from "./index";

/*
 * Configure Enzyme
 */
configure({ adapter: new Adapter() });

describe("heading snapshots", (): void => {
    generateSnapshots(examples as SnapshotTestSuite<HeadingProps>);
});

describe("heading", (): void => {
    test("should have a displayName that matches the component name", () => {
        expect((MSFTHeading as any).name).toBe(MSFTHeading.displayName);
    });

    test("should not throw if managedClasses are not provided", () => {
        expect(() => {
            shallow(<MSFTHeading tag={HeadingTag.h1} />);
            shallow(<MSFTHeading tag={HeadingTag.h1} size={HeadingSize._1} />);
        }).not.toThrow();
    });

    test("should accept unhandledProps", () => {
        const handledProps: HeadingHandledProps = {
            tag: HeadingTag.h1,
            size: HeadingSize._1,
        };

        const unhandledProps: HeadingUnhandledProps = {
            "aria-hidden": true,
        };

        const props: HeadingHandledProps & HeadingUnhandledProps = {
            ...handledProps,
            ...unhandledProps,
        };

        const rendered: any = mount(<Heading {...props} />);

        expect(rendered.find(handledProps.tag).prop("aria-hidden")).toEqual(true);
    });

    test("should render the correct `tag` when `tag` prop is passed", () => {
        const rendered: any = mount(<MSFTHeading tag={HeadingTag.h3} />);

        expect(rendered.exists(HeadingTag.h3)).toBe(true);
    });
});
