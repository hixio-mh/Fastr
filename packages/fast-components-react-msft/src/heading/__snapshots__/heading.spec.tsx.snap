// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading snapshots Heading: 0 1`] = `
<h1
  className="typography-0-1-8 typography__1-0-1-9 heading-0-1-1 heading__1-0-1-2"
>
  Heading test string
</h1>
`;

exports[`heading snapshots Heading: 1 1`] = `
<h1
  className="typography-0-1-25 typography__1-0-1-26 heading-0-1-18 heading__1-0-1-19"
  data-sketch-symbol="Heading 1"
>
  Heading test string
</h1>
`;

exports[`heading snapshots Heading: 2 1`] = `
<h2
  className="typography-0-1-42 typography__2-0-1-44 heading-0-1-35 heading__2-0-1-37"
  data-sketch-symbol="Heading 2"
>
  Heading test string
</h2>
`;

exports[`heading snapshots Heading: 3 1`] = `
<h3
  className="typography-0-1-59 typography__3-0-1-62 heading-0-1-52 heading__3-0-1-55"
  data-sketch-symbol="Heading 3"
>
  Heading test string
</h3>
`;

exports[`heading snapshots Heading: 4 1`] = `
<h4
  className="typography-0-1-76 typography__4-0-1-80 heading-0-1-69 heading__4-0-1-73"
  data-sketch-symbol="Heading 4"
>
  Heading test string
</h4>
`;

exports[`heading snapshots Heading: 5 1`] = `
<h5
  className="typography-0-1-93 typography__5-0-1-98 heading-0-1-86 heading__5-0-1-91"
  data-sketch-symbol="Heading 5"
>
  Heading test string
</h5>
`;

exports[`heading snapshots Heading: 6 1`] = `
<h6
  className="typography-0-1-110 typography__6-0-1-116 heading-0-1-103 heading__6-0-1-109"
  data-sketch-symbol="Heading 6"
>
  Heading test string
</h6>
`;
